src/join_mesh.py

```diff
@@ -25,260 +15,264 @@
-
-        part_point_data = part_mesh.GetPointData()
-        num_arrays = part_point_data.GetNumberOfArrays()
-
-        if joined_data_arrays is None:
-            joined_data_arrays = []
-            for j in range(num_arrays):
-                newArr = vtk.vtkDoubleArray()
-                newArr.SetName(part_point_data.GetArrayName(j))
-                joined_data_arrays.append(newArr)
-
-        for j in range(num_arrays):
-            array_name = part_point_data.GetArrayName(j)
-            logging.debug("Merging from file {} dataname {}".format(fname, array_name))
-            array_data = part_point_data.GetArray(array_name)
-            join_arr = joined_data_arrays[j]
-            join_arr.SetNumberOfComponents(array_data.GetNumberOfComponents())
-            join_arr.InsertTuples(join_arr.GetNumberOfTuples(), array_data.GetNumberOfTuples(), 0, array_data)
-
-        for i in range(part_mesh.GetNumberOfCells()):
-            cell = part_mesh.GetCell(i)
-            vtkCell = vtk.vtkGenericCell()
-            vtkCell.SetCellType(cell.GetCellType())
-            idList = vtk.vtkIdList()
-            for j in range(cell.GetNumberOfPoints()):
-                idList.InsertNextId(cell.GetPointId(j) + offset)
-            vtkCell.SetPointIds(idList)
-            joined_cell_types.append(cell.GetCellType())
-            joined_cells.InsertNextCell(vtkCell)
-
-        offset += part_mesh.GetNumberOfPoints()
-
-    if len(joined_cell_types) != 0:
-        joined_mesh.SetCells(joined_cell_types, joined_cells)
-    joined_mesh.SetPoints(joined_points)
-    for data_array in joined_data_arrays:
-        joined_mesh.GetPointData().AddArray(data_array)
-
-    return joined_mesh
-
-
-def join_mesh_recovery(prefix: str, partitions: int, recoveryPath: str):
-    """
-    Partition merge with full recovery
-
-    This recovers the original mesh.
-    """
-    logging.info("Starting full mesh recovery")
-    recovery = json.load(open(recoveryPath, "r"))
-    cells = recovery["cells"]
-    size = recovery["size"]
-    cell_types = recovery["cell_types"]
-
-    logging.debug("Original Mesh contains {} points".format(size))
-    logging.debug("{} Cells discarded during partitioning".format(len(cells)))
-
-    # Initialize Joined Mesh
-    joined_mesh = vtk.vtkUnstructuredGrid()
-    joined_points = vtk.vtkPoints()
-    joined_points.SetNumberOfPoints(size)
-    joined_data_arrays = None
-    joined_cells = vtk.vtkCellArray()
-    joined_cell_types = []
-
-    for i in range(partitions):
-        global_ids = []
-
-        fname = prefix + "_" + str(i) + ".vtu"
-        reader = vtk.vtkXMLUnstructuredGridReader()
-        reader.SetFileName(fname)
-        reader.Update()
-        part_mesh = reader.GetOutput()
-        part_point_data = part_mesh.GetPointData()
-        num_arrays = part_point_data.GetNumberOfArrays()
-
-        # Prepare DataArrays
-        if joined_data_arrays is None:
-            joined_data_arrays = []
+class MeshJoiner:
```

Comment:
I guess the class name should include 'ASTE' somehow. We can decide on this later when we actually discuss the renaming. We should also add a description to the class as usually done in python (maybe the one from below?).

Reply:
Class description added. Naming is not so important ATM Let's discuss while renaming scripts.